packages/aws-cdk/test/api/hotswap/hotswap-test-setup.ts

```diff
@@ -0,0 +1,85 @@
+import * as cxapi from '@aws-cdk/cx-api';
+import { CloudFormation } from 'aws-sdk';
+import * as lambda from 'aws-sdk/clients/lambda';
+import * as stepfunctions from 'aws-sdk/clients/stepfunctions';
+import { testStack, TestStackArtifact } from '../../util';
+import { Template } from '../../../lib/api/util/cloudformation';
+import { MockSdkProvider } from '../../util/mock-sdk';
+import { FakeCloudformationStack } from './fake-cloudformation-stack';
+import { DeployStackResult } from '../../../lib';
+import * as deployments from '../../../lib/api/hotswap-deployments';
+
+const STACK_NAME = 'withouterrors';
+export const STACK_ID = 'stackId';
+
+let mockSdkProvider: MockSdkProvider;
+let currentCfnStack: FakeCloudformationStack;
+const currentCfnStackResources: CloudFormation.StackResourceSummary[] = [];
+
+export function setupHotswapTests() {
+  jest.resetAllMocks();
+  mockSdkProvider = new MockSdkProvider({ realSdk: false });
+  // clear the array
+  currentCfnStackResources.splice(0);
+  mockSdkProvider.stubCloudFormation({
+    listStackResources: ({ StackName: stackName }) => {
+      if (stackName !== STACK_NAME) {
+        throw new Error(`Expected Stack name in listStackResources() call to be: '${STACK_NAME}', but received: ${stackName}'`);
+      }
+      return {
+        StackResourceSummaries: currentCfnStackResources,
+      };
+    },
+  });
+
+  currentCfnStack = new FakeCloudformationStack({
+    stackName: STACK_NAME,
+    stackId: STACK_ID,
+  });
+}
+
+export function setUpdateFunctionCodeMock(mockUpdateLambdaCode: (input: lambda.UpdateFunctionCodeRequest) => lambda.FunctionConfiguration) {
+  mockSdkProvider.stubLambda({
+    updateFunctionCode: mockUpdateLambdaCode,
+  });
+}
+
+export function setUpdateStateMachineMock(mockUpdateMachineDefinition:
+(input: stepfunctions.UpdateStateMachineInput) =>
+stepfunctions.UpdateStateMachineOutput) {
+  mockSdkProvider.stubStepFunctions({
+    updateStateMachine: mockUpdateMachineDefinition,
+  });
+}
+
+export function cdkStackArtifactOf(testStackArtifact: Partial<TestStackArtifact> = {}): cxapi.CloudFormationStackArtifact {
+  return testStack({
+    stackName: STACK_NAME,
+    ...testStackArtifact,
+  });
+}
+
+export function pushStackMocks(...items: CloudFormation.StackResourceSummary[]) {
```

Comment:
This name is weird - this is not dealing with mocks, but with CloudFormation Resource Summaries. I think this should be renamed.